### PR TITLE
Added JUnit tests for exception and model packages

### PR DESCRIPTION
**Pull request by sonja-ct**
This PR JUnit tests for classes in exception and model packages in rest-api/resources module:

 - exception package:
RestApiRuntimeException class
SessionNotPopulatedException class

- model package
ByteArrayParam class
CountResult class
DateParam class
EntityId class
MetricType class
ScopeId class
StorableEntityId class

Also it contains JUnit tests for class AbstractKapuaResource.

Signed-off-by: Sonja <sonja.matic@endava.com>

**Related Issue**
/

**Description of the solution adopted**
/

**Screenshots**
/

**Any side note on the changes made**
/

### CODE REVIEW

**Review by LeoNerdoG — APPROVED** [review]
hey @Coduz,
I have checked this PR and it looks OK. This is part of "rest api unit tests" that we have been talking about (achieving code coverage for rest-api module through unit tests).
After all these PRs are merged and we have framework for Rest Api testing in place, we will add high-level/integration rest api tests, that will check the actual fucntionality.
Please take a look and merge when you can.

**Review by LeoNerdoG — APPROVED** [review]

File: rest-api/core/src/test/java/org/eclipse/kapua/app/api/core/exception/RestApiRuntimeExceptionTest.java
Comment (Coduz) [inline rest-api/core/src/test/java/org/eclipse/kapua/app/api/core/exception/RestApiRuntimeExceptionTest.java]: Please make use of `@Test(expected = NullPointerException.class)`

File: rest-api/core/src/test/java/org/eclipse/kapua/app/api/core/exception/RestApiRuntimeExceptionTest.java
Comment (Coduz) [inline rest-api/core/src/test/java/org/eclipse/kapua/app/api/core/exception/RestApiRuntimeExceptionTest.java]: Please make use of `@Test(expected = NullPointerException.class)`

File: rest-api/core/src/test/java/org/eclipse/kapua/app/api/core/exception/RestApiRuntimeExceptionTest.java
Comment (Coduz) [inline rest-api/core/src/test/java/org/eclipse/kapua/app/api/core/exception/RestApiRuntimeExceptionTest.java]: Please make use of `@Test(expected = NullPointerException.class)`

File: rest-api/core/src/test/java/org/eclipse/kapua/app/api/core/model/DateParamTest.java
Comment (Coduz) [inline rest-api/core/src/test/java/org/eclipse/kapua/app/api/core/model/DateParamTest.java]: DateParam.getDate() returns a Date.
So this is always true

File: rest-api/core/src/test/java/org/eclipse/kapua/app/api/core/model/DateParamTest.java
Comment (Coduz) [inline rest-api/core/src/test/java/org/eclipse/kapua/app/api/core/model/DateParamTest.java:35]: Please make use of `@Test(expected = NullPointerException.class)`

File: rest-api/core/src/test/java/org/eclipse/kapua/app/api/core/model/DateParamTest.java
Comment (Coduz) [inline rest-api/core/src/test/java/org/eclipse/kapua/app/api/core/model/DateParamTest.java:42]: This is the correct way of expecting exception in a test! 😁

File: rest-api/core/src/test/java/org/eclipse/kapua/app/api/core/model/EntityIdTest.java
Comment (Coduz) [inline rest-api/core/src/test/java/org/eclipse/kapua/app/api/core/model/EntityIdTest.java]: Please make use of `@Test(expected = NullPointerException.class)`

File: rest-api/core/src/test/java/org/eclipse/kapua/app/api/core/model/ScopeIdTest.java
Comment (Coduz) [inline rest-api/core/src/test/java/org/eclipse/kapua/app/api/core/model/ScopeIdTest.java]: Please make use of `@Test(expected = NullPointerException.class)`

**Review by Coduz — CHANGES_REQUESTED** [review]

**Review by LeoNerdoG — APPROVED** [review]

**Comment by sonja-ct** [comment]
Hey @Coduz I made requested changes, so you can merge this when you have time. Thanks

**Comment by sonja-ct** [comment]
Hey @Coduz I made requested changes, so you can merge this when you have time. Thanks

**Pull request merged**
